src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -22,10 +23,30 @@
 )
 
 
+def _deprecated_memory_arg(shared_memory: bool, share_inputs: bool) -> bool:
```

Comment:
is it OK to deprecate API in patch release?
I suppose - no

maybe we need to backport only feature itself? while deprecation is only since 2023.1

Reply:
As users are getting new flags which introduction resulted in deprecation of others... I do not see why prompting the same message for them is not valid.

In case of both 2023.0 and 2023.1, we only tell that older flag will be removed and it is not advised to be used in combination with newer ones. Especially this part is important, there is some difference in the behavior itself:
> Value of `shared_memory` is going to override `share_inputs` value. Please use only `share_inputs` explicitly.